inference-engine/src/transformations/src/transformations/common_optimizations/pad_fusion.cpp

```diff
@@ -0,0 +1,202 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/pad_fusion.hpp"
+#include "transformations/utils/utils.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/opsets/opset5.hpp>
+
+using namespace ngraph;
+
+NGRAPH_RTTI_DEFINITION(pass::PadFusion, "PadFusion", 0);
+
+template <typename T>
+static bool can_be_fused(const std::shared_ptr<opset5::Pad>& pad, const std::shared_ptr<T>& node,
+                         const std::shared_ptr<opset5::Constant>& pad_value_const,
+                         const std::shared_ptr<opset5::Constant>& pads_begin, const std::shared_ptr<opset5::Constant>& pads_end) {
+    if (!pad || pad->get_pad_mode() != op::PadMode::CONSTANT)
+        return false;
+    if (!node)
+        return false;
+
+    if (!pad_value_const)
+        return false;
+    auto pad_value = pad_value_const->cast_vector<float>()[0];
+    if (pad_value != 0.0f)
+        return false;
+
+    if (!pads_begin || !is_vector(pads_begin->get_shape()))
+        return false;
+    if (!pads_end || !is_vector(pads_end->get_shape()))
+        return false;
+    if (node->get_pads_begin().size() != shape_size(pads_begin->get_shape()) - 2)
+        return false;
+    if (node->get_pads_end().size() != shape_size(pads_end->get_shape()) - 2)
+        return false;
+
+    return true;
+}
+
+template <typename T>
+static std::tuple<Shape, Shape> new_pad_values(const std::shared_ptr<opset5::Constant>& pads_begin,
+                                               const std::shared_ptr<opset5::Constant>& pads_end,
+                                               const std::shared_ptr<T>& node) {
+    auto node_pads_begin = node->get_pads_begin();
+    auto node_pads_end = node->get_pads_end();
+    auto pads_begin_val = pads_begin->cast_vector<size_t>();
+    auto pads_end_val = pads_end->cast_vector<size_t>();
+
+    std::transform(node_pads_begin.begin(), node_pads_begin.end(),
+                   pads_begin_val.begin() + 2, node_pads_begin.begin(),
+                   [] (size_t a, size_t b) -> size_t { return a + b; });
+    std::transform(node_pads_end.begin(), node_pads_end.end(),
+                   pads_end_val.begin() + 2, node_pads_end.begin(),
+                   [] (size_t a, size_t b) -> size_t { return a + b; });
+
+    return std::make_tuple(node_pads_begin, node_pads_end);
+}
+
+NGRAPH_RTTI_DEFINITION(pass::PadFusionAvgPool, "PadFusionAvgPool", 0);
+
+pass::PadFusionAvgPool::PadFusionAvgPool() {
+    auto data_pattern = pattern::any_input();
+    auto pads_begin_pattern = pattern::wrap_type<opset5::Constant>();
+    auto pads_end_pattern = pattern::wrap_type<opset5::Constant>();
+    auto pad_value_pattern = pattern::wrap_type<opset5::Constant>();
+    auto pad_node_pattern = pattern::wrap_type<opset5::Pad>({data_pattern, pads_begin_pattern, pads_end_pattern, pad_value_pattern});
```

Comment:
Check hat pad has exactly one consumer
```suggestion
    auto pad_node_pattern = pattern::wrap_type<opset5::Pad>({data_pattern, pads_begin_pattern, pads_end_pattern, pad_value_pattern}, pattern::consumers_count(1));
```